app/bundles/LeadBundle/Form/Type/FieldType.php

```diff
@@ -219,6 +220,34 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         )
                     )
                 );
+            } elseif (in_array($type, array('datetime', 'date', 'time'))) {
+                $constraints = array();
+                if ($type === 'datetime') {
+                    $constraints = array(new Assert\Callback(function ($object, ExecutionContextInterface $context) {
+                        if (\DateTime::createFromFormat('Y-m-d H:i', $object) === false) {
+                            $context->buildViolation('mautic.lead.datetime.invalid')->addViolation();
+                        }
+                    }));
+                } elseif ($type === 'date') {
+                    $constraints = array(new Assert\Date);
```

Comment:
Can we add a custom error message for this? Took me a minute to figure out what format it wanted because the default is just a generic "this is an invalid date"